include/ddsrouter/participant/implementations/rtps/impl/DiscoveryServerParticipant.ipp

```diff
@@ -353,32 +337,57 @@ void DiscoveryServerParticipant<ConfigurationType>::enable_tls(
     descriptor->tls_config.add_option(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSOptions::NO_SSLV2); // not safe
 
-    // CA certificate
-    descriptor->tls_config.verify_file = get_param(tls_config, TLS_CA_TAG);
     // Perform verification of the server
     descriptor->tls_config.add_verify_mode(
         eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_PEER);
 
-    if (client_only)
+    if (client)
     {
-        // Fail verification if the server has no certificate
-        descriptor->tls_config.add_verify_mode(
-            eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_FAIL_IF_NO_PEER_CERT);
+        if (!tls_configuration->can_be_client())
+        {
+            logError(DDSROUTER_DISCOVERYSERVER_PARTICIPANT,
+                    "TLS Configuration expected a Client configuration.");
+            throw ConfigurationException("TLS Configuration expected a Client configuration.");
+        }
+        else
+        {
+            std::shared_ptr<security::TlsConfigurationClient> tls_configuration_ =
+                    std::dynamic_pointer_cast<security::TlsConfigurationClient>(tls_configuration);
+
+            // Fail verification if the server has no certificate
+            descriptor->tls_config.add_verify_mode(
+                eprosima::fastdds::rtps::TCPTransportDescriptor::TLSConfig::TLSVerifyMode::VERIFY_FAIL_IF_NO_PEER_CERT);
+
+            // CA certificate
+            descriptor->tls_config.verify_file = tls_configuration_->certificate_authority_file();
```

Comment:
I believe this should be set for both clients and servers, since no descriptor is configured as client if a descriptor configured as server has already been pushed.